v3/src/components/tool-shelf/tool-shelf.tsx

```diff
@@ -1,68 +1,121 @@
+      hint: t("DG.ToolButtonData.optionMenu.toolTip")
+    },
+    {
+      icon: <HelpIcon className="icon-help"/>,
+      label: t("DG.ToolButtonData.help.title"),
+      hint: t("DG.ToolButtonData.help.toolTip")
+    },
+    {
+      icon: <GuideIcon className="icon-guid"/>,
+      label: t("DG.ToolButtonData.guideMenu.title"),
+      hint: t("DG.ToolButtonData.guideMenu.toolTip")
+    }
+  ]
+
   const keys = getTileComponentKeys()
   const entries = keys.map(key => getTileComponentInfo(key))
                       .filter(info => info?.shelf != null) as IShelfTileComponentInfo[]
   entries.sort((a, b) => a.shelf.position - b.shelf.position)
 
+  function handleTileButtonClick(tileType: string) {
+    content?.createOrShowTile?.(tileType)
+  }
+
+  function handleRightButtonClick(entry: IRightButtonEntry) {
+    if (entry.onClick) {
+      entry.onClick()
+    }
+    else {
+      labelToast(entry)
+    }
+  }
+
   return (
-    <HStack className='tool-shelf' alignContent='center' data-testid='tool-shelf'>
+    <Flex className='tool-shelf' alignContent='center' data-testid='tool-shelf'>
       <Flex className="tool-shelf-component-buttons">
         {entries.map((entry, idx) => {
           if (!entry) return null
-          const { type, shelf: { ButtonComponent = ToolShelfButton, label, hint } } = entry
+          const { type, shelf: { ButtonComponent = ToolShelfTileButton, labelKey, hintKey } } = entry
+          const label = t(labelKey)
+          const hint = t(hintKey)
           return (
             ButtonComponent
-              ? <ButtonComponent tileType={type} key={`${type}-${idx}`} label={label} hint={hint} content={content}/>
+              ? <ButtonComponent tileType={type} key={`${type}-${idx}`} label={label} hint={hint}
+                    onClick={handleTileButtonClick}/>
               : null
           )
         })}
       </Flex>
-    </HStack>
-  )
-}
-
-export interface IToolShelfButtonProps {
-  tileType: string
-  label: string
-  hint: string
-  content?: IDocumentContentModel
-}
-
-export const ToolShelfButton = ({tileType, label, hint, content}: IToolShelfButtonProps) => {
-  const Icon = getTileComponentIcon(tileType)
-
-  return (
-    <Box
-      as='button'
-      bg='white'
-      title={t(hint)}
-      onClick={() => content?.createOrShowTile?.(tileType)}
-      data-testid={`tool-shelf-button-${t(label)}`}
-      className="tool-shelf-button"
-      _hover={{ boxShadow: '1px 1px 1px 0px rgba(0, 0, 0, 0.5)' }}
-      // :active styling is in css to override Chakra default
-    >
-      <>
-        {Icon && <Icon />}
-        <Tag className='tool-shelf-tool-label'>{t(label)}</Tag>
-      </>
-    </Box>
+      <Spacer/>
+      <Flex className="tool-shelf-right-buttons">
+        {rightButtons.map(entry => {
+          const { className, icon, label, hint } = entry
+          return <ToolShelfButton key={entry.label} className={className} icon={icon} label={label} hint={hint}
```

Comment:
This very likely doesn't matter much, but is there a reason you used `entry.label` for the `key` attribute, and `label` for the `label` attribute? If I understand correctly, they're the same thing.

Reply:
Nope -- will fix